exercises/shared-api-gateway/ITOps.ViewModelComposition/CompositionHandler.cs

```diff
@@ -30,7 +32,17 @@ public static async Task<(dynamic ViewModel, int StatusCode)> HandleGetRequest(H
 
                 foreach (var appender in interceptors.OfType<IViewModelAppender>())
                 {
-                    pendingTasks.Add(appender.Append(viewModel, routeData, context.Request.Query));
+                    pendingTasks.Add(appender.Append(viewModel, routeData, context.Request.Query)
+                        .ContinueWith(t =>
+                        {
+                            t.Exception.Handle(ex =>
+                            {
+                                var logger = loggerFactory.CreateLogger(appender.GetType());
+                                //adjust to emit the correct information
```

Comment:
What does this comment mean?

Reply:
Because I’m not sure that Exception.ToString() is the right thing to do in this case. I need some @danielmarbach -fu 😄

Reply:
That was my first attempt and I like it much more. However at that point we’ve lost the connection between the failed Task and the failing Appender/Handler. We’re using the appender/handler type to create a specific logger

Reply:
@mauroservienti @adamralph what do you think about this https://github.com/Particular/Workshop/pull/264